### ISSUE TITLE
CloudFront  - No endpoint for service cloudfront in region ''

### ISSUE THREAD

**Issue by dnmfarrell** [pplu/aws-sdk-perl#113]
I don't think CloudFront has regions, but if a region argument isn't passed, this error is thrown:

```
No endpoint for service cloudfront in region ''

Trace begun at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/API/EndpointResolver.pm line 83
Paws::API::EndpointResolver::_construct_endpoint('Paws::CloudFront=HASH(0x5488ab8)') called at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/API/EndpointResolver.pm line 14 
Paws::API::EndpointResolver::__ANON__('Paws::CloudFront=HASH(0x5488ab8)') called at reader Paws::CloudFront::_endpoint_info (defined at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/API/EndpointResolver.pm line 16) line 6
Paws::CloudFront::_endpoint_info('Paws::CloudFront=HASH(0x5488ab8)') called at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/API/EndpointResolver.pm line 44 
Paws::API::EndpointResolver::__ANON__('Paws::CloudFront=HASH(0x5488ab8)') called at reader Paws::CloudFront::_api_endpoint (defined at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/API/EndpointResolver.pm line 46) line 6
Paws::CloudFront::_api_endpoint('Paws::CloudFront=HASH(0x5488ab8)') called at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/Net/RestXmlCaller.pm line 161
Paws::Net::RestXmlCaller::prepare_request_for_call('Paws::CloudFront=HASH(0x5488ab8)', 'Paws::CloudFront::CreateInvalidation=HASH(0x54f5338)') called at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/Net/Caller.pm line 18 
Paws::Net::Caller::send_request('Paws::Net::Caller=HASH(0x34639b8)', 'Paws::CloudFront=HASH(0x5488ab8)', 'Paws::CloudFront::CreateInvalidation=HASH(0x54f5338)') called at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/Net/RetryCallerRole.pm line 18 
Paws::Net::RetryCallerRole::do_call('Paws::Net::Caller=HASH(0x34639b8)', 'Paws::CloudFront=HASH(0x5488ab8)', 'Paws::CloudFront::CreateInvalidation=HASH(0x54f5338)') called at /home/dfarrell/.plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/Paws/CloudFront.pm line 58 
Paws::CloudFront::CreateInvalidation('Paws::CloudFront=HASH(0x5488ab8)', 'DistributionId', 'ECFFL7U0PVO9A', 'InvalidationBatch', 'HASH(0x5402970)')
```

**Comment by pplu** [comment]
Thanks for the info! I think that passing `us-east-1` is going to work (it's the region that AWS uses for "global services" like CloudFront). I have to look at the region resolver to diagnose this better, since at first look, CloudFront has the correct endpoint information to work without a region.

**Comment by pplu** [comment]
@rcrews @dnmfarrell: refactoring around for 0.33 I finally found what was wrong with CloudFront: https://github.com/pplu/aws-sdk-perl/tree/fix/global-services

This will most probably be merged into release 0.33. If you can test it out, it'd be great!

**Comment by pplu** [comment]
Just pushed 0.33 to CPAN with this issue solved
